addons/web/static/tests/views/fields/one2many_field.test.js

```diff
@@ -12868,3 +12868,57 @@ test("onchange on x2many add and delete x2m record, returning to initial state",
+    Turtle._views["form, false"] = `<form><field name="name"/></form>`;
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        arch: `<form><field name="turtles"><tree><field name="name"/></tree></field></form>`,
+        resId: 1,
+    });
+    expect(".o_field_widget[name=turtles] .o_data_row").toHaveCount(1);
+    expect(".o_data_cell").toHaveText("donatello");
+    await contains(queryFirst(".o_field_widget[name=turtles] .o_data_cell")).click();
+    expect(".o_dialog .o_form_view").toHaveCount(1);
+    expect(".o_dialog .modal-header .o_expand_button").toHaveCount(1);
+    await contains(".o_dialog .modal-header .btn-close").click();
+    await contains(".o_field_widget[name=turtles] .o_field_x2many_list_row_add a").click();
+    expect(".o_dialog .o_form_view").toHaveCount(1);
+    expect(".o_dialog .modal-header .o_expand_button").toHaveCount(0);
+    await contains("[name='name'] input").edit("new turtle");
+    await contains(".o_dialog .o_form_button_save").click();
+    expect(".o_field_widget[name=turtles] .o_data_row").toHaveCount(2);
+    await contains(queryLast(".o_field_widget[name=turtles] .o_data_cell")).click();
+    expect(".o_dialog .o_form_view").toHaveCount(1);
+    expect(".o_dialog .modal-header .o_expand_button").toHaveCount(0);
+    await contains(".o_dialog .modal-header .btn-close").click();
+    await clickSave();
+    await contains(queryLast(".o_field_widget[name=turtles] .o_data_cell")).click();
+    expect(".o_dialog .o_form_view").toHaveCount(1);
+    expect(".o_dialog .modal-header .o_expand_button").toHaveCount(1);
+});
+
+test.tags("desktop")("new record - expand record in dialog", async () => {
+    Turtle._views["form, false"] = `<form><field name="name"/></form>`;
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        arch: `<form><field name="turtles"><tree><field name="name"/></tree></field></form>`,
+    });
+    expect(".o_field_widget[name=turtles] .o_data_row").toHaveCount(0);
+    await contains(".o_field_widget[name=turtles] .o_field_x2many_list_row_add a").click();
+    expect(".o_dialog .o_form_view").toHaveCount(1);
+    expect(".o_dialog .modal-header .o_expand_button").toHaveCount(0);
+    await contains("[name='name'] input").edit("new turtle");
+    await contains(".o_dialog .o_form_button_save").click();
+    expect(".o_field_widget[name=turtles] .o_data_row").toHaveCount(1);
+    await contains(queryFirst(".o_field_widget[name=turtles] .o_data_cell")).click();
+    expect(".o_dialog .o_form_view").toHaveCount(1);
+    expect(".o_dialog .modal-header .o_expand_button").toHaveCount(0);
+    await contains(".o_dialog .modal-header .btn-close").click();
+    await clickSave();
+    await contains(queryLast(".o_field_widget[name=turtles] .o_data_cell")).click();
+    expect(".o_dialog .o_form_view").toHaveCount(1);
+    expect(".o_dialog .modal-header .o_expand_button").toHaveCount(1);
+});
```

Comment:
not sure this one really brings anything new. As the other test requires the existing model to be saved for the newly created one2many to be expandable

Reply:
Indeed, I would keep only the first one. However, you only assert the presence of the button, but you don't click on it (to assert that the expected action is executed, with a mockService("action"))